### DISCUSSION TITLE
Usage with other querying libraries

### DISCUSSION THREAD

**Discussion by ghost** [trpc/trpc#1756] [Q&A]
Hi thank you so much for awesome library!

I am using redux toolkit in my project, and it comes with rtk query, so my question is is there any way I can automatically generate 
rtk query api using trpc types?

I am quite new to rtk query and did not dive into trpc source code yet. It would be great if you could show some direction which I might head to start.

**Comment by KATT** [comment]
I think you should just use react-query but you can dive in here if you wanna see how it works https://github.com/trpc/trpc/blob/main/packages/react/src/createReactQueryHooks.tsx

**Reply by KATT** [reply]
Also see #1014

**Reply by KATT** [reply]
This is nothing I will add official support for so you're on your own with the implementation, I don't have the bandwidth to help out.

**Comment by spirobel** [comment]
very good point. The nice thing about rtk query is that it automatically handles caching and saving of the data. There needs to be some of the shelf to solution to this kind of problem. maybe integrate it with rtk

**Comment by epeterson320** [comment]
@spirobel I was able to get it working with the following. Although eventually I realized I didn't need it. Regarding caching, the TRPC docs let demonstrate how to do it via web standards with the `cache-control` header so it might be worth it to evaluate if you really need javascript caching logic.

```tsx
import type { AppRouter } from "@repo/api/src/appRouter";
import type { inferProcedureOutput } from "@trpc/server";
import { createApi } from "@reduxjs/toolkit/query/react";
import { createTRPCProxyClient } from "@trpc/client";

type Status = inferProcedureOutput<AppRouter["status"]["get"]>;

const trpcClient = createTRPCProxyClient<AppRouter>(/* ... */);

const api = createApi({
  reducerPath: "api",
  baseQuery: (trpcResult: Promise<unknown>) =>
    trpcResult.then(data => ({ data })).catch(error => ({ error })),
  endpoints: (builder) => ({
    getStatus: builder.query<Status, void>({
      query: trpcClient.status.get.query,
    }),
  }),
});

const { useGetStatusQuery } = api;

const StatusPage = () => {
  const {
    data: status, // types correctly
  } = useGetStatusQuery();

  return <p>{status.someField}</p>;
};
```

**Reply by jonathanrgreene** [reply]
Doesn't seem to work. I always end up with any type.
